### PR TITLE
Allow custom options in GOVUK.analytics.trackPageview

### PR DESCRIPTION
**Pull request by colinrotherham**
Currently the `GOVUK.analytics.trackPageview()` method only supports the `transport` Google Analytics fieldsObject property, which puts a limit on its usefulness.

This limitation makes the following overrides unavailable:
`hitCallback`
`sessionControl`

This pull request allows any options object to be used.

For example, the current session can now be reset when returning to a specific page (perhaps when a kiosk session times out):
```javascript
GOVUK.analytics.trackPageview('/', 'Home', {
  sessionControl: 'start'
})
```

### CODE REVIEW

**Comment by colinrotherham** [comment]
Hi @dsingleton, @nickcolley,

I've improved the way options can be set on `GOVUK.analytics.trackPageview()`, would you mind taking a look for me in this pull request?

The third `options`parameter I've extended isn't documented in **analytics.md** but it would be useful to others. Shall I add this or would you rather keep it as a hidden feature?

File: javascripts/govuk/analytics/google-analytics-universal-tracker.js [hunk 1]
```diff
@@ -33,24 +34,25 @@
 
   // https://developers.google.com/analytics/devguides/collection/analyticsjs/pages
   GoogleAnalyticsUniversalTracker.prototype.trackPageview = function(path, title, options) {
-    var options = options || {};
+    var options;
```
Comment (robinwhittleton) [inline javascripts/govuk/analytics/google-analytics-universal-tracker.js]: Don’t think we need this line right? `options` is coming in as an argument and if it’s not supplied `if (typeof options === "object")` will just fail as `options` will equal `'undefined'`.

**Comment by nickcolley** [comment]
Hey @colinrotherham 

Thanks for the PR! I had a look over the code it looks good, I think it'd be good to add the note to the docs if you have time.. 👍

File: javascripts/govuk/analytics/google-analytics-universal-tracker.js (hunk 1 above)
Comment (colinrotherham) [inline javascripts/govuk/analytics/google-analytics-universal-tracker.js]: Good point. I saw it as a simplification, but missed that you were re-declaring `options`.

Let's tidy this up.

Comment (colinrotherham) [inline javascripts/govuk/analytics/google-analytics-universal-tracker.js]: The `trackEvent` method was doing the same.

Now fixed: https://github.com/alphagov/govuk_frontend_toolkit/pull/332/files#diff-8b93af583c1a7ca5e9c48a89ebce2b38L63

Comment (nickcolley) [inline javascripts/govuk/analytics/google-analytics-universal-tracker.js]: This would be caught when we add linting 👍

**Comment by colinrotherham** [comment]
Thanks @nickcolley.

Spotted by @robinwhittleton were two places `options` was being redeclared (both `trackEvent()` and `trackPageview()` which are now sorted.

**Comment by robinwhittleton** [comment]
So this looks conceptually fine to me, but I haven’t tested it on an actual site and I’m not going to have time to this week. @nickcolley, is this something you could do?

**Comment by colinrotherham** [comment]
Hi @nickcolley,

Do you still plan to merge this in?

Thanks

**Comment by nickcolley** [comment]
@colinrotherham I haven't had the time to test this sorry Colin will try to ASAP but hopefully someone else can review, will ping others too..

**Comment by robinwhittleton** [comment]
Tested this with a few different inputs and it’s working as expected, so I’m going to merge. It would be good to get this tested with analytics people on GOV.UK before static is updated to include it.

**Pull request merged**